subsys/bluetooth/host/smp.c

```diff
@@ -2693,6 +2693,10 @@ static uint8_t get_auth(struct bt_smp *smp, uint8_t auth)
 		auth &= ~BT_SMP_AUTH_KEYPRESS;
 	}
 
+#if defined(CONFIG_BT_CLASSIC)
+	auth |= BT_SMP_AUTH_CT2;
+#endif /* CONFIG_BT_CLASSIC */
```

Comment:
I'm not so sure that this is needed, since CT2 is already included in the definition of `BT_SMP_AUTH_DEFAULT` which is passed as input to this function.

Reply:
There is a special case that the `auth_req` of pairing response of IUT will be calculated by using the value of `auth_req` of pairing request of peer. 
`rsp->auth_req = get_auth(smp, req->auth_req);`
If the `CT2` flag is not set in `auth_req` of pairing request, the `CT2` flag will also not set in `auth_req` of pairing response.

Reply:
Right, but if the remote side didn't set it, then setting it locally will not help, right? I.e. doesn't using CT2 require that both the request and response set the flag?

Reply:
I took a look at the exact text in the core spec, and it states:
> The CT2 field is a 1-bit flag that shall be set to 1 upon transmission to indicate support for the h7 function. See sections 2.4.2.4 and 2.4.2.5.

Based on that it does indeed sound like we should unconditionally always set the flag. I also don't see anything BR/EDR related about this policy, so I don't understand why you have this protected by `#if defined(CONFIG_BT_CLASSIC)`?

Anyway, if it should always be set, I think I'd instead modify the call to `get_auth()` to be:
```c
	rsp->auth_req = get_auth(smp, req->auth_req | BT_SMP_AUTH_CT2);
```

Reply:
That said, from a pure practical perspective it doesn't seem like this makes any difference, since setting `SMP_FLAG_CT2` requires both the request and response to have set the flag, i.e. it seems to me like the core spec is asking for more than what's actually necessary.

Reply:
> I took a look at the exact text in the core spec, and it states:
> 
> > The CT2 field is a 1-bit flag that shall be set to 1 upon transmission to indicate support for the h7 function. See sections 2.4.2.4 and 2.4.2.5.
> 
> Based on that it does indeed sound like we should unconditionally always set the flag. I also don't see anything BR/EDR related about this policy, so I don't understand why you have this protected by `#if defined(CONFIG_BT_CLASSIC)`?
> 
> Anyway, if it should always be set, I think I'd instead modify the call to `get_auth()` to be:
> 
> ```c
> 	rsp->auth_req = get_auth(smp, req->auth_req | BT_SMP_AUTH_CT2);
> ```

I think this bit is valid only when dual mode is supported.

Reply:
> Right, but if the remote side didn't set it, then setting it locally will not help, right? I.e. doesn't using CT2 require that both the request and response set the flag?

The `CT2` flag should not depend on peer device.
And there is a test case `SM/PER/SCCT/BV-08-C` for this case.

Reply:
> I think this bit is valid only when dual mode is supported.

The SMP specification doesn't explicitly state that, but it makes of course practical sense since this is specifically about Cross-Transport Key Derivation.